### DISCUSSION TITLE
RaspiBlitz in a Proxmox VM

### DISCUSSION THREAD

**Discussion by openoms** [raspiblitz/raspiblitz#4282] [Show and tell]
Just had a successful RaspiBlitz install in a Proxmox VM.  Install went very smooth and had no issues.  I'm running Proxmox VE 8.1.3 and used Debian 12.2.0 for the VM.  Hardware is a Ryzen 5700g, 32GB DDR4 RAM, and a Samsung 990Pro 4TB NVME SSD.

Steps taken
1. Setup the VM in Proxmox to install Debian, CPU set to Host, and allotted a 64GB drive on the 990Pro to emulate the SD card.
2. Started the VM and installed Debian with no GUI.
3. Booted into the VM, switched to root and installed sudo with ' apt install sudo '
4. Downloaded the RaspiBlitz build script with ' wget https://raw.githubusercontent.com/rootzoll/raspiblitz/dev/build_sdcard.sh '
5. Ran the build script with ' sudo bash build_sdcard.sh -f true -b dev -d headless -t false -w off '
6. When the script finished the install I shut down the VM.
7. Configured a second 1.5TB SSD on the 990Pro for the VM to emulate the data drive and restarted the VM.
8. SSH into the VM and the setup took off just like on the Raspberry Pi 4.  

So far I have Electrs, Mempool, JoininBox, and C-Lightning running successfully.  I have Zeus connected to the node as well.  Happy to say this install was completely painless and it runs MUCH faster than on the Pi.  Just thought I'd share my experience.

_Originally posted by @Silarous in https://github.com/raspiblitz/raspiblitz/issues/2104#issuecomment-1828956785_

**Comment by alex27riva** [comment]
Build failed for me, missing PGP key

```
# importing key of rustyrussell
gpg: directory '/home/bitcoin/.gnupg' created
gpg: keybox '/home/bitcoin/.gnupg/pubring.kbx' created
gpg: key D9200E6CD1ADB8F1: 37 signatures not checked due to missing keys
pub   rsa4096 2011-10-18 [SC]
      15EE8D6CAB0E7F0CF999BFCBD9200E6CD1ADB8F1
uid                      Rusty Russell <rusty@rustcorp.com.au>
sub   rsa4096 2011-10-18 [E]

gpg: key D9200E6CD1ADB8F1: 37 signatures not checked due to missing keys
gpg: /home/bitcoin/.gnupg/trustdb.gpg: trustdb created
gpg: key D9200E6CD1ADB8F1: public key "Rusty Russell <rusty@rustcorp.com.au>" imported
gpg: Total number processed: 1
gpg:               imported: 1
gpg: no ultimately trusted keys found
# running: git verify-tag v23.11

gpg: Signature made Tue 28 Nov 2023 07:04:16 PM CET
gpg:                using EDDSA key 653B19F33DF7EFF3E9D1C94CC3F21EE387FF4CD2
gpg: Can't check signature: No public key
# goodSignature(0)
# correctKey(0)

# BUILD FAILED --> PGP verification not OK / signature(0) verify(0)
```

**Reply by openoms** [reply]
the CLN upgrade was premature, will be done in PR #4281
the dev branch should be working now if you try again

**Comment by alex27riva** [comment]
Build failed again?
```
> mempool-backend@2.4.0 tsc
> ./node_modules/typescript/bin/tsc

# blitz.display.sh set-display lcd 
baseimage(debian)
# old(headless)
# new(lcd)
err='baseimage not supported'

```

**Reply by Silarous** [reply]
I received that same message at the end of the build.  It should still work.  Just reboot and make sure you SSH in with the username ' admin ' and password ' raspiblitz '

**Reply by alex27riva** [reply]
Are you using a physical drive or internal storage for the data?

**Reply by Silarous** [reply]
My main drive and data drive are both on the 990Pro NVME that is in the m.2 port on the motherboard.  Also passed through a 8gb USB stick for the emergency backup.

**Reply by openoms** [reply]
would suggest to not use fatpack for your build. The base image does not need to contain all that, on amd64 compiling things doesn't take that much.
use the option: `-f false`

**Comment by Silarous** [comment]
Just updated the VM install to the latest RaspiBlitz release v1.11.0.  Steps taken

1. Selected Update -> Release in the RaspiBlitz menu, downloaded the Lightning backup, and shut the machine down.
2. Added a CD drive to the VM and loaded the Debian 12 Bookworm ISO. (This Raspiblitz release is used with Debian 12)
3. Changed the boot order to boot from the CD drive first.
4. Started the VM and installed Debian to my smaller 64GB drive writing over the old data and shutdown the VM. (Don't write over your data drive!)
5. Removed the CD drive from the VM.
6. Started the VM, switched to root and installed sudo/wget.  "apt install sudo wget"
7.  Downloaded the build script.  "wget https://raw.githubusercontent.com/rootzoll/raspiblitz/dev/build_sdcard.sh"
8. Ran the build script.  "sudo bash build_sdcard.sh -f false -b dev -d headless -t false -w off"
9. Restarted the VM and SSH into it with admin:raspiblitz.
10. It found my data drive and began the restore process!

One thing to note, I have a USB drive added to this VM for a SCB/Emergency-Backup.  On my first attempt at this upgrade, I left the USB drive attached and when I first booted into the new RaspiBlitz install, it thought my USB drive was my data drive, formatted it, and wouldn't restore my old system state as a result.  I repeated the process with the USB drive detached and it found the correct 1.5TB data drive.  Once it was done, I reattached the USB drive, enabled the USB backup, and all was well.  Overall a painless upgrade!

**Reply by Silarous** [reply]
Updated to release v1.11.1 following the same instructions successfully.  Had an issue the first time I ran the build script and it failed after setting up resolveconf.  It stopped resolving DNS queries.  No nameservers were identified in /etc/resolv.conf.  Added my local nameservers and ran the build script again.  Steps taken

1. Selected Update -> Release in the RaspiBlitz menu, downloaded the Lightning backup, and shut the machine down.
2. Added a CD drive to the VM and loaded the Debian 12.6.0 Bookworm ISO. (This Raspiblitz release is used with Debian 12.6.0)
3. Removed USB backup drive if you have one.
4. Changed the boot order to boot from the CD drive first.
5. Started the VM and installed Debian to my smaller 64GB drive writing over the old data and shutdown the VM. (Don't write over your data drive!)
6. Removed the CD drive from the VM.
7. Started the VM, switched to root and installed sudo/wget. "apt install sudo wget"
8. Downloaded the build script. "wget https://raw.githubusercontent.com/rootzoll/raspiblitz/dev/build_sdcard.sh"
9. Ran the build script. "sudo bash build_sdcard.sh -f false -b dev -d headless -t false -w off"
10. First attempt failed after setting up resolveconf.
11. Ran the command   echo "nameserver 8.8.8.8" | sudo tee /etc/resolv.conf   to add a nameserver. (change the IP to your DNS)
12. Ran the build script again. "sudo bash build_sdcard.sh -f false -b dev -d headless -t false -w off"
13. Restarted the VM and SSH into it with admin:raspiblitz.
14. It found my data drive and began the restore process.
15. Once completely restored and running, plugged the USB backup drive back in and re-enabled it in the menu.

**Reply by tscharly** [reply]
"echo "nameserver 8.8.8.8" | sudo tee /etc/resolv.conf" saved my day!

**Comment by adevoss** [comment]
Same steps (`sudo ./build_sdcard.sh -f true -b dev -d headless -t false -w off` ) but then #4682
Earlier releases were successful.

Edit:
With `sudo ./build_sdcard.sh -f false -b dev -d headless -t false -w off` the setup works, resulted in v1.11.2, but still #4569.

**Reply by rootzoll** [reply]
adding 8.8.8.8 will be part of v1.11.3rc1 - see: https://github.com/raspiblitz/raspiblitz/blob/755c9bcf43bd90975ee52420fb3a755da72ba443/build_sdcard.sh#L358-L359

**Comment by Silarous** [comment]
Upgraded from 1.11.1 to 1.11.4 following the same steps as before.  Only issue I ran into was cRTL not able to talk to CLN.  When I would enter the RTL sign-in page, after inputting my password and clicking "Dashboard" I was given the error "Core Lightning Get Info Failed Due To Missing Rune!".  I found the rune file in /mnt/hdd/app-data/rtl/cRTL/rune_for_cRTL and inside it just listed "LIGHTNING_RUNE=" with no rune information.  I ran "lightning-cli showrunes", copied my rune from there, and pasted it into the run_for_cRTL file as LIGHTNING_RUNE="runedata".  Made sure the quotes are around the rune.  Restarted cRTL and it connected fine.  Before doing this I uninstalled RTL and then switched it back on again but that didn't help the issue.  Steps taken to upgrade

1. Selected Update -> Release in the RaspiBlitz menu, downloaded the Lightning backup, and shut the machine down.
2. Added a CD drive to the VM and loaded the Debian 12.6.0 Bookworm ISO. (This Raspiblitz release is used with Debian 12.6.0)
3. Removed my USB backup drive.
4. Changed the boot order to boot from the CD drive first.
5. Started the VM and installed Debian to my smaller 64GB drive writing over the old data and shutdown the VM. (Don't write over your data drive!)
6. Removed the CD drive from the VM.
7. Started the VM, switched to root and installed sudo/wget. "apt install sudo wget"
8. Downloaded the build script. "wget https://raw.githubusercontent.com/rootzoll/raspiblitz/dev/build_sdcard.sh"
9. Ran the build script. "sudo bash build_sdcard.sh -f false -b dev -d headless -t false -w off"
10. Restarted the VM and SSH into it with admin:raspiblitz.
11. It found my data drive and began the restore process.
12. Once completely restored and running, plugged the USB backup drive back in and re-enabled it in the menu.

**Comment by SepticFrank** [comment]
Would something like this work on a mini pc similar to what the Start 9 server runs on? https://store.start9.com/products/server-one

Maybe a user friendly setup as well?

I have one, but I don't use it because I prefer running raspiblitz after all (would love to convert it into a blitz). 

But I really liked that the node synced in about 2 days because much better the hardware is. And you can put Bisq offers up 24/7 without having to keep a laptop open all the time. They use some sort of web portal to access Bisq on the server. Being able to do that while running raspiblitz would be amazing.

**Reply by Silarous** [reply]
I don't see why not.  Should be able to install Debian directly to the Start9 box and then run the build script.  It's just a mini PC after all.

**Comment by ursus-caerulus** [comment]
Dear all,
I keep running into the same problem when trying to install raspiblitz in proxmox. I used the steps listed above.
After installing I get:

***********************************************************
Stop signal detectecd - OK ready for manual provision.
If your ready for shutdown use the following command:
release --> for an official release
release -quick --> during development
REMOVE AUDIO DEVICE from VM before next boot up.
***********************************************************

Running the provision results in the same error.
There is nog audio device configured/attached.
No difference between fatpack or no fatpack install.
No difference in using the dev buildsd / version 1.11 buildsd.

Any tips?

Edit:
Apologies, didn't search the closed topics. Found the answer in #4899
